browser/browser_context_keyed_service_factories.cc

```diff
@@ -104,6 +106,10 @@ void EnsureBrowserContextKeyedServiceFactoriesBuilt() {
 #endif
 
   PermissionLifetimeManagerFactory::GetInstance();
+
+  if (base::FeatureList::IsEnabled(skus::features::kSkusFeature)) {
```

Comment:
this check actually shouldn't be here after all, the convention in upstream is to return nullptr for the service when the feature is disabled

Reply:
actually upstream does both, but we have been returning nullptr in other service factories and I think we should follow that convention

Reply:
it's definitely simpler if the feature check stays internal to the service factory so we don't have to spread them all over the code

Reply:
Fixed in https://github.com/brave/brave-core/pull/10847/commits/bc08735f11dc33351fe3845cef9a29959b4e6eb4